packit_service/worker/build/build_helper.py

```diff
@@ -377,12 +428,27 @@ def _report(
         The status reporting should be done through this method
         so we can extend it in subclasses easily.
         """
-        self.status_reporter.report(
-            description=description,
-            state=state,
-            url=url,
-            check_names=check_names,
-        )
+        if self.is_gitlab_instance and not self.is_reporting_allowed:
+            # "Packit-User" does not have access to project.
+
+            self.request_project_access()
+            build_states = [
+                CommitStatus.error,
+                CommitStatus.success,
+                CommitStatus.failure,
+            ]
+
+            # We are only commenting final states to avoid multiple comments for a build
+            # Ignoring all other states eg. pending, running
+            if state in build_states and check_names is not None:
+                self.status_reporter.report_status_by_comment(state, url, check_names)
```

Comment:
```suggestion
            final_commit_states = [
                CommitStatus.error,
                CommitStatus.success,
                CommitStatus.failure,
            ]

            # We are only commenting final states to avoid multiple comments for a build
            # Ignoring all other states eg. pending, running
            if state in final_commit_states and check_names is not None:
                self.status_reporter.report_status_by_comment(state, url, check_names)
```